libkbfs/config_local.go

```diff
@@ -1459,3 +1460,19 @@ func (c *ConfigLocal) SetKBFSService(k *KBFSService) {
 	}
 	c.kbfsService = k
 }
+
+// RootNodeWrappers implements the Config interface for ConfigLocal.
+func (c *ConfigLocal) RootNodeWrappers() []func(Node) Node {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	ret := make([]func(Node) Node, len(c.rootNodeWrappers))
+	copy(ret, c.rootNodeWrappers)
+	return ret
+}
+
+// AddRootNodeWrapper implements the Config interface for ConfigLocal.
+func (c *ConfigLocal) AddRootNodeWrapper(f func(Node) Node) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.rootNodeWrappers = append(c.rootNodeWrappers, f)
```

Comment:
Shouldn't you also update the node cache's root node wrappers?

Reply:
Sounds good, will do.

Reply:
Oops, I realized I put my comment above on the wrong comment.  It belonged on the thread above about returning a copy of `c.rootNodeWrappers`.

As for updating the node cache's root node wrappers -- that's hard to do, because there's not just one node cache, there is one for each FBO.  That's why I commented the interface definition for `AddRootNodeWrapper` like this:

```go
// This will only apply to TLFs that are first accessed after `AddRootNodeWrapper` is called.
```

Seems like it's not worth supporting something more complicated that adds them after startup, but let me know if you see a need for that.

Reply:
I guess with that comment it's fine.